### PR TITLE
Add moky88 keyboard

### PR DESCRIPTION
**Pull request by VertorWang**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/moky/moky88/info.json [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+          "solid_reactive_multicross": true,
+          "solid_reactive_nexus": true,
+          "solid_reactive_multinexus": true,
+          "splash": true,
+          "multisplash": true,
+          "solid_splash": true,
+          "solid_multisplash": true
+      },
+      "layout": [
+            {"flags": 4, "matrix": [0, 0], "x": 0  , "y":  0},
```
Comment (zvecr) [inline keyboards/moky/moky88/info.json:93]: You declare 90 LEDs but only have 88 elements here.

File: keyboards/moky/moky88/info.json [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+          {"pin_a": "B6", "pin_b": "B7"}
+      ]
+  },
+  "rgb_matrix": {
+      "driver": "aw20216",
+      "max_brightness": 108,
+      "val_steps": 27,
+      "speed_steps": 64,
+      "animations": {
+          "solid_color": true,
```
Comment (zvecr) [inline keyboards/moky/moky88/info.json:47]:
```suggestion
```

File: keyboards/moky/moky88/info.json [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+  "encoder": {
+      "rotary": [
+          {"pin_a": "B6", "pin_b": "B7"}
+      ]
+  },
+  "rgb_matrix": {
+      "driver": "aw20216",
+      "max_brightness": 108,
+      "val_steps": 27,
+      "speed_steps": 64,
```
Comment (zvecr) [inline keyboards/moky/moky88/info.json:45]:
```suggestion
```

File: keyboards/moky/moky88/moky88.c [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+const aw_led g_aw_leds[RGB_MATRIX_LED_COUNT] = {
```
Comment (zvecr) [inline keyboards/moky/moky88/moky88.c:19]:
```suggestion
const aw20216s_led_t PROGMEM g_aw20216s_leds[AW20216S_LED_COUNT] = {
```

File: keyboards/moky/moky88/config.h [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+#define DRIVER_2_EN B9
+
+#define DRIVER_COUNT 2
+#define DRIVER_1_LED_TOTAL 65
+#define DRIVER_2_LED_TOTAL 25
+#define RGB_MATRIX_LED_COUNT (DRIVER_1_LED_TOTAL + DRIVER_2_LED_TOTAL)
+
+#define RGB_DISABLE_WHEN_USB_SUSPENDED
+#define RGB_MATRIX_FRAMEBUFFER_EFFECTS
+#define RGB_MATRIX_KEYPRESSES
```
Comment (zvecr) [inline keyboards/moky/moky88/config.h:28]:
```suggestion
```

File: keyboards/moky/moky88/info.json [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+            {"flags": 4, "matrix": [5, 2], "x": 28 , "y": 64},
+            {"flags": 4, "matrix": [5, 6], "x": 84 , "y": 64},
+            {"flags": 4, "matrix": [5,10], "x": 140, "y": 64},
+            {"flags": 4, "matrix": [5,11], "x": 154, "y": 64},
+            {"flags": 4, "matrix": [5,12], "x": 168, "y": 64},
+            {"flags": 4, "matrix": [5,13], "x": 182, "y": 64},
+            {"flags": 4, "matrix": [5,14], "x": 196, "y": 64},
+            {"flags": 4, "matrix": [5,15], "x": 210, "y": 64},
+            {"flags": 4, "matrix": [5,16], "x": 224, "y": 64}
+        ]
```
Comment (zvecr) [inline keyboards/moky/moky88/info.json:181]:
```suggestion
        ],
        "sleep": true
```

File: keyboards/moky/moky88/info.json [hunk 7]
```diff
@@ -0,0 +1,277 @@
+{
+  "manufacturer": "moky",
+  "keyboard_name": "moky88",
+  "maintainer": " ",
+  "url": "",
+  "processor": "WB32FQ95",
+  "bootloader": "wb32-dfu",
+  "bootmagic": {
+      "matrix": [0, 0]
+  },
```
Comment (zvecr) [inline keyboards/moky/moky88/info.json:10]:
```suggestion
```

File: keyboards/moky/moky88/info.json [hunk 8]
```diff
@@ -0,0 +1,277 @@
+{
+  "manufacturer": "moky",
+  "keyboard_name": "moky88",
+  "maintainer": " ",
```
Comment (zvecr) [inline keyboards/moky/moky88/info.json:4]: should not be a blank string

File: keyboards/moky/moky88/config.h [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+#define SPI_MISO_PIN B4
+
+/* Flash */
+#define EXTERNAL_FLASH_SPI_SLAVE_SELECT_PIN C12
+
+/* RGB Driver */
+#define DRIVER_1_CS D2
+#define DRIVER_2_CS B8
+#define DRIVER_1_EN B9
+#define DRIVER_2_EN B9
```
Comment (zvecr) [inline keyboards/moky/moky88/config.h:19]:
```suggestion
#define AW20216S_CS_PIN_1 D2
#define AW20216S_CS_PIN_2 B8
#define AW20216S_EN_PIN_1 B9
#define AW20216S_EN_PIN_2 B9
```

File: keyboards/moky/moky88/keymaps/via/keymap.c [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+    [3] = LAYOUT(  /* FN */
+        _______, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_MUTE, KC_PSCR, KC_SCRL, KC_PAUSE,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, RGB_SPD, RGB_SPI, RGB_TOG, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, RGB_SAD, RGB_SAI, RGB_MOD, _______, _______, _______,
+        _______, TO(0),   TO(2),   _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,                   RGB_VAI,
+        _______, _______, _______,                            _______,                            _______, _______, _______, _______, RGB_HUD, RGB_VAD, RGB_HUI),
+};
+
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
```
Comment (zvecr) [inline keyboards/moky/moky88/keymaps/via/keymap.c:42]:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
```

File: keyboards/moky/moky88/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+    [3] = LAYOUT(  /* FN */
+        _______, KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_MUTE, KC_PSCR, KC_SCRL, KC_PAUSE,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, RGB_SPD, RGB_SPI, RGB_TOG, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, RGB_SAD, RGB_SAI, RGB_MOD, _______, _______, _______,
+        _______, TO(0),   TO(2),   _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,                   RGB_VAI,
+        _______, _______, _______,                            _______,                            _______, _______, _______, _______, RGB_HUD, RGB_VAD, RGB_HUI),
+};
+
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
```
Comment (zvecr) [inline keyboards/moky/moky88/keymaps/default/keymap.c:42]:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
```

File: keyboards/moky/moky88/readme.md [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+
+Make example for this keyboard (after setting up your build environment):
+
+    make moky/moky88:default
+
+Flashing example for this keyboard:
+
+    make moky/moky88:default:flash
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```
Comment (zvecr) [inline keyboards/moky/moky88/readme.md:17]: Missing bootloader section per [template](https://github.com/qmk/qmk_firmware/blob/master/data/templates/keyboard/readme.md)

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/moky/moky88/info.json [hunk 13]
```diff
@@ -0,0 +1,277 @@
...
+            {"flags": 4, "matrix": [5,11], "x": 154, "y": 64},
+            {"flags": 4, "matrix": [5,12], "x": 168, "y": 64},
+            {"flags": 4, "matrix": [5,13], "x": 182, "y": 64},
+            {"flags": 4, "matrix": [5,14], "x": 196, "y": 64},
+            {"flags": 4, "matrix": [5,15], "x": 210, "y": 64},
+            {"flags": 4, "matrix": [5,16], "x": 224, "y": 64}
+        ]
+  },
+  "layouts": {
+      "LAYOUT": {
```
Comment (dunk2k) [inline keyboards/moky/moky88/info.json:184]:
```suggestion
  "community_layouts": ["tkl_f13_ansi"],
  "layouts": {
      "LAYOUT_tkl_f13_ansi": {
```

File: keyboards/moky/moky88/keymaps/default/keymap.c [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+
+    [2] = LAYOUT(  /* Base */
+        KC_ESC,  KC_BRMD, KC_BRMU, _______, _______, _______, _______, KC_MPRV, KC_MPLY, KC_MNXT, KC_MUTE, KC_VOLD, KC_VOLU, KC_MUTE, KC_PSCR, KC_SCRL, KC_PAUSE,
+        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_INS,  KC_HOME, KC_PGUP,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_DEL,  KC_END,  KC_PGDN,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,
+        KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT,                   KC_UP,
+        KC_LCTL, KC_LALT, KC_LGUI,                            KC_SPC,                             KC_RGUI, MO(3),   KC_APP,  KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT),
+
+    [3] = LAYOUT(  /* FN */
```
Comment (dunk2k) [inline keyboards/moky/moky88/keymaps/default/keymap.c:33]:
```suggestion
    [0] = LAYOUT_tkl_f13_ansi( /* Base */
        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_MUTE, KC_PSCR, KC_SCRL, KC_PAUSE,
        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_INS,  KC_HOME, KC_PGUP,
        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_DEL,  KC_END,  KC_PGDN,
        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,
        KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT,                   KC_UP,
        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, MO(1),   KC_APP,  KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT),

    [1] = LAYOUT_tkl_f13_ansi( /* Base */
        _______, KC_MSEL, KC_VOLD, KC_VOLU, KC_MUTE, KC_MSTP, KC_MPRV, KC_MPLY, KC_MNXT, KC_MAIL, KC_WHOM, KC_CALC, KC_WSCH, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, RGB_SPD, RGB_SPI, RGB_TOG, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, RGB_SAD, RGB_SAI, RGB_MOD, _______, _______, _______,
        _______, TO(0),   TO(2),   _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,                   RGB_VAI,
        _______, GU_TOGG, _______,                            _______,                            _______, _______, _______, _______, RGB_HUD, RGB_VAD, RGB_HUI),

    [2] = LAYOUT_tkl_f13_ansi(  /* Base */
        KC_ESC,  KC_BRMD, KC_BRMU, _______, _______, _______, _______, KC_MPRV, KC_MPLY, KC_MNXT, KC_MUTE, KC_VOLD, KC_VOLU, KC_MUTE, KC_PSCR, KC_SCRL, KC_PAUSE,
        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_INS,  KC_HOME, KC_PGUP,
        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_DEL,  KC_END,  KC_PGDN,
        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,
        KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT,                   KC_UP,
        KC_LCTL, KC_LALT, KC_LGUI,                            KC_SPC,                             KC_RGUI, MO(3),   KC_APP,  KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT),

    [3] = LAYOUT_tkl_f13_ansi(  /* FN */
```

File: keyboards/moky/moky88/keymaps/via/keymap.c [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+
+    [2] = LAYOUT(  /* Base */
+        KC_ESC,  KC_BRMD, KC_BRMU, _______, _______, _______, _______, KC_MPRV, KC_MPLY, KC_MNXT, KC_MUTE, KC_VOLD, KC_VOLU, KC_MUTE, KC_PSCR, KC_SCRL, KC_PAUSE,
+        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_INS,  KC_HOME, KC_PGUP,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_DEL,  KC_END,  KC_PGDN,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,
+        KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT,                   KC_UP,
+        KC_LCTL, KC_LALT, KC_LGUI,                            KC_SPC,                             KC_RGUI, MO(3),   KC_APP,  KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT),
+
+    [3] = LAYOUT(  /* FN */
```
Comment (dunk2k) [inline keyboards/moky/moky88/keymaps/via/keymap.c:33]:
```suggestion
    [0] = LAYOUT_tkl_f13_ansi( /* Base */
        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_MUTE, KC_PSCR, KC_SCRL, KC_PAUSE,
        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_INS,  KC_HOME, KC_PGUP,
        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_DEL,  KC_END,  KC_PGDN,
        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,
        KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT,                   KC_UP,
        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, MO(1),   KC_APP,  KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT),

    [1] = LAYOUT_tkl_f13_ansi( /* Base */
        _______, KC_MSEL, KC_VOLD, KC_VOLU, KC_MUTE, KC_MSTP, KC_MPRV, KC_MPLY, KC_MNXT, KC_MAIL, KC_WHOM, KC_CALC, KC_WSCH, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, RGB_SPD, RGB_SPI, RGB_TOG, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, RGB_SAD, RGB_SAI, RGB_MOD, _______, _______, _______,
        _______, TO(0),   TO(2),   _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,                   RGB_VAI,
        _______, GU_TOGG, _______,                            _______,                            _______, _______, _______, _______, RGB_HUD, RGB_VAD, RGB_HUI),

    [2] = LAYOUT_tkl_f13_ansi(  /* Base */
        KC_ESC,  KC_BRMD, KC_BRMU, _______, _______, _______, _______, KC_MPRV, KC_MPLY, KC_MNXT, KC_MUTE, KC_VOLD, KC_VOLU, KC_MUTE, KC_PSCR, KC_SCRL, KC_PAUSE,
        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_INS,  KC_HOME, KC_PGUP,
        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_DEL,  KC_END,  KC_PGDN,
        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,
        KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT,                   KC_UP,
        KC_LCTL, KC_LALT, KC_LGUI,                            KC_SPC,                             KC_RGUI, MO(3),   KC_APP,  KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT),

    [3] = LAYOUT_tkl_f13_ansi(  /* FN */
```

**Review by dunk2k — CHANGES_REQUESTED** [review]
add Community Layout support

**Comment by dunk2k** [comment]
directory `keymaps/default/` is missing a `rules.mk` file containing `ENCODER_MAP_ENABLE = yes`

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/moky/moky88/readme.md (hunk 12 above)
Comment (VertorWang) [reply keyboards/moky/moky88/readme.md]: Add a key to enter the bootloader, and add instructions to the readme documentation

**Comment by VertorWang** [comment]
> add Community Layout support

How do I add community layout support

**Comment by VertorWang** [comment]
I have solved the problem, please check and merge. thank you.

File: keyboards/moky/moky88/info.json [hunk 16]
```diff
@@ -0,0 +1,272 @@
...
+            {"matrix": [5,1], "x": 1.25, "y": 5.25, "w": 1.25 },
+            {"matrix": [5,2], "x": 2.5, "y": 5.25, "w": 1.25 },
+            {"matrix": [5,6], "x": 3.75, "y": 5.25, "w": 6.5 },
+            {"matrix": [5,10], "x": 10.25, "y": 5.25, "w": 1.25 },
+            {"matrix": [5,11], "x": 11.5, "y": 5.25, "w": 1.25 },
+            {"matrix": [5,12], "x": 12.75, "y": 5.25, "w": 1.25 },
+            {"matrix": [5,13], "x": 14, "y": 5.25, "w": 1.25 },
+            {"matrix": [5,14], "x": 15.75, "y": 5.25 },
+            {"matrix": [5,15], "x": 16.75, "y": 5.25 },
+            {"matrix": [5,16], "x": 17.75, "y": 5.25 }
```
Comment (dunk2k) [inline keyboards/moky/moky88/info.json:268]: adjust key widths and spacing to match `readme.md` image
```suggestion
            {"matrix": [0,1], "x": 1.25, "y": 0 },
            {"matrix": [0,2], "x": 2.25, "y": 0 },
            {"matrix": [0,3], "x": 3.25, "y": 0 },
            {"matrix": [0,4], "x": 4.25, "y": 0 },
            {"matrix": [0,5], "x": 5.5, "y": 0 },
            {"matrix": [0,6], "x": 6.5, "y": 0 },
            {"matrix": [0,7], "x": 7.5, "y": 0 },
            {"matrix": [0,8], "x": 8.5, "y": 0 },
            {"matrix": [0,9], "x": 9.75, "y": 0 },
            {"matrix": [0,10], "x": 10.75, "y": 0 },
            {"matrix": [0,11], "x": 11.75, "y": 0 },
            {"matrix": [0,12], "x": 12.75, "y": 0 },
            {"matrix": [0,13], "x": 14, "y": 0 },
            {"matrix": [0,14], "x": 15.25, "y": 0 },
            {"matrix": [0,15], "x": 16.25, "y": 0 },
            {"matrix": [0,16], "x": 17.25, "y": 0 },
            {"matrix": [1,0], "x": 0, "y": 1.25 },
            {"matrix": [1,1], "x": 1, "y": 1.25 },
            {"matrix": [1,2], "x": 2, "y": 1.25 },
            {"matrix": [1,3], "x": 3, "y": 1.25 },
            {"matrix": [1,4], "x": 4, "y": 1.25 },
            {"matrix": [1,5], "x": 5, "y": 1.25 },
            {"matrix": [1,6], "x": 6, "y": 1.25 },
            {"matrix": [1,7], "x": 7, "y": 1.25 },
            {"matrix": [1,8], "x": 8, "y": 1.25 },
            {"matrix": [1,9], "x": 9, "y": 1.25 },
            {"matrix": [1,10], "x": 10, "y": 1.25 },
            {"matrix": [1,11], "x": 11, "y": 1.25 },
            {"matrix": [1,12], "x": 12, "y": 1.25 },
            {"matrix": [1,13], "x": 13, "y": 1.25, "w": 2 },
            {"matrix": [1,14], "x": 15.25, "y": 1.25 },
            {"matrix": [1,15], "x": 16.25, "y": 1.25 },
            {"matrix": [1,16], "x": 17.25, "y": 1.25 },
            {"matrix": [2,0], "x": 0, "y": 2.25, "w": 1.5 },
            {"matrix": [2,1], "x": 1.5, "y": 2.25 },
            {"matrix": [2,2], "x": 2.5, "y": 2.25 },
            {"matrix": [2,3], "x": 3.5, "y": 2.25 },
            {"matrix": [2,4], "x": 4.5, "y": 2.25 },
            {"matrix": [2,5], "x": 5.5, "y": 2.25 },
            {"matrix": [2,6], "x": 6.5, "y": 2.25 },
            {"matrix": [2,7], "x": 7.5, "y": 2.25 },
            {"matrix": [2,8], "x": 8.5, "y": 2.25 },
            {"matrix": [2,9], "x": 9.5, "y": 2.25 },
            {"matrix": [2,10], "x": 10.5, "y": 2.25 },
            {"matrix": [2,11], "x": 11.5, "y": 2.25 },
            {"matrix": [2,12], "x": 12.5, "y": 2.25 },
            {"matrix": [2,13], "x": 13.5, "y": 2.25, "w": 1.5 },
            {"matrix": [2,14], "x": 15.25, "y": 2.25 },
            {"matrix": [2,15], "x": 16.25, "y": 2.25 },
            {"matrix": [2,16], "x": 17.25, "y": 2.25 },
            {"matrix": [3,0], "x": 0, "y": 3.25, "w": 1.75 },
            {"matrix": [3,1], "x": 1.75, "y": 3.25 },
            {"matrix": [3,2], "x": 2.75, "y": 3.25 },
            {"matrix": [3,3], "x": 3.75, "y": 3.25 },
            {"matrix": [3,4], "x": 4.75, "y": 3.25 },
            {"matrix": [3,5], "x": 5.75, "y": 3.25 },
            {"matrix": [3,6], "x": 6.75, "y": 3.25 },
            {"matrix": [3,7], "x": 7.75, "y": 3.25 },
            {"matrix": [3,8], "x": 8.75, "y": 3.25 },
            {"matrix": [3,9], "x": 9.75, "y": 3.25 },
            {"matrix": [3,10], "x": 10.75, "y": 3.25 },
            {"matrix": [3,11], "x": 11.75, "y": 3.25 },
            {"matrix": [3,13], "x": 12.75, "y": 3.25, "w": 2.25 },
            {"matrix": [4,0], "x": 0, "y": 4.25, "w": 2.25 },
            {"matrix": [4,2], "x": 2.25, "y": 4.25 },
            {"matrix": [4,3], "x": 3.25, "y": 4.25 },
            {"matrix": [4,4], "x": 4.25, "y": 4.25 },
            {"matrix": [4,5], "x": 5.25, "y": 4.25 },
            {"matrix": [4,6], "x": 6.25, "y": 4.25 },
            {"matrix": [4,7], "x": 7.25, "y": 4.25 },
            {"matrix": [4,8], "x": 8.25, "y": 4.25 },
            {"matrix": [4,9], "x": 9.25, "y": 4.25 },
            {"matrix": [4,10], "x": 10.25, "y": 4.25 },
            {"matrix": [4,11], "x": 11.25, "y": 4.25 },
            {"matrix": [4,13], "x": 12.25, "y": 4.25, "w": 2.75 },
            {"matrix": [4,15], "x": 16.25, "y": 4.25 },
            {"matrix": [5,0], "x": 0, "y": 5.25, "w": 1.25 },
            {"matrix": [5,1], "x": 1.25, "y": 5.25, "w": 1.25 },
            {"matrix": [5,2], "x": 2.5, "y": 5.25, "w": 1.25 },
            {"matrix": [5,6], "x": 3.75, "y": 5.25, "w": 6.25 },
            {"matrix": [5,10], "x": 10, "y": 5.25, "w": 1.25 },
            {"matrix": [5,11], "x": 11.25, "y": 5.25, "w": 1.25 },
            {"matrix": [5,12], "x": 12.5, "y": 5.25, "w": 1.25 },
            {"matrix": [5,13], "x": 13.75, "y": 5.25, "w": 1.25 },
            {"matrix": [5,14], "x": 15.25, "y": 5.25 },
            {"matrix": [5,15], "x": 16.25, "y": 5.25 },
            {"matrix": [5,16], "x": 17.25, "y": 5.25 }
```

File: keyboards/moky/moky88/info.json [hunk 17]
```diff
@@ -0,0 +1,272 @@
...
+          ]
+      }
+  }
```
Comment (dunk2k) [inline keyboards/moky/moky88/info.json:271]: formatting
```suggestion
    }
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Comment by dunk2k** [comment]
> > add Community Layout support
> 
> How do I add community layout support

Already have, when respectively named suggestion was accepted 😉

File: keyboards/moky/moky88/rules.mk [hunk 18]
```diff
@@ -0,0 +1 @@
+# This file intentionally left blank
```
Comment (dunk2k) [inline keyboards/moky/moky88/rules.mk:1]:
```suggestion
# This file intentionally left blank

```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/moky/moky88/readme.md [hunk 19]
```diff
@@ -0,0 +1,26 @@
+# moky88
+![image](https://github.com/VertorWang/qmk_firmware/assets/96033026/15564003-4d7c-4fd9-8259-5202695cc76b)
```
Comment (waffle87) [inline keyboards/moky/moky88/readme.md:2]: Please rehost this image on Imgur.

**Review by drashna — APPROVED** [review]

**Review by waffle87 — APPROVED** [review]

**Pull request merged**